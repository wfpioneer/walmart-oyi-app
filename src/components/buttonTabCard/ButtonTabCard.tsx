import React from 'react';
import { View } from 'react-native';
import COLOR from '../../themes/Color';
import Button from '../buttons/Button';
import styles from './ButtonTabCard.style';

interface ButtonBottomTabProps{
  leftTitle: string;
  onLeftPress: () => void;
  rightTitle: string;
  onRightPress: () => void;
}

export const ButtonBottomTab = (props: ButtonBottomTabProps): JSX.Element => {
  const {
    leftTitle, onLeftPress, rightTitle, onRightPress
  } = props;
  return (
    <View style={styles.mainBarContainer}>
      <Button
        style={styles.buttonAlign}
        title={leftTitle}
        titleColor={COLOR.MAIN_THEME_COLOR}
        type={Button.Type.SOLID_WHITE}
        onPress={() => onLeftPress()}
      />
      <Button
        style={styles.buttonAlign}
        title={rightTitle}
        type={Button.Type.PRIMARY}
        onPress={() => onRightPress()}
      />
    </View>
  );
};
