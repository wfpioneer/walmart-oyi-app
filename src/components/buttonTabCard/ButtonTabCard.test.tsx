import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { ButtonBottomTab } from './ButtonTabCard';

describe('ButtonTabCard Component', () => {
  it('Renders the Button Bottom Tab component', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <ButtonBottomTab
        leftTitle="LeftButtonTitle"
        onLeftPress={jest.fn()}
        onRightPress={jest.fn()}
        rightTitle="RightButtonTitle"
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
