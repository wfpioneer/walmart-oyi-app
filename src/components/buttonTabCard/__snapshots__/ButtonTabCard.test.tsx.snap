// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonTabCard Component Renders the Button Bottom Tab component 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "backgroundColor": "#ffffffff",
        "elevation": 10,
        "flexDirection": "row",
        "height": 80,
        "justifyContent": "space-evenly",
      },
      Object {
        "height": 80,
      },
    ]
  }
>
  <Button
    backgroundColor="#0071e9"
    disabled={false}
    height={45}
    onPress={[Function]}
    radius={5}
    style={
      Object {
        "alignSelf": "center",
        "width": "45%",
      }
    }
    title="LeftButtonTitle"
    titleColor="#0071e9"
    titleFontSize={16}
    titleFontWeight="normal"
    type={2}
    width={-1}
  />
  <Button
    backgroundColor="#0071e9"
    disabled={false}
    height={45}
    onPress={[Function]}
    radius={5}
    style={
      Object {
        "alignSelf": "center",
        "width": "45%",
      }
    }
    title="RightButtonTitle"
    titleColor="#ffffffff"
    titleFontSize={16}
    titleFontWeight="normal"
    type={1}
    width={-1}
  />
</View>
`;
